src/main.rs

```diff
@@ -0,0 +1,482 @@
+                    (note, cents)
+                }
+            };
+
+            min_amount += BLOCK_SIZE / 2;
+
+            terminal.draw(|f| {
+                let main_layout = Layout::default()
+                    .direction(Direction::Vertical)
+                    .margin(1)
+                    .constraints(
+                        [
+                            Constraint::Min(7),
+                            Constraint::Percentage(20),
+                            Constraint::Percentage(80),
+                        ]
+                        .as_ref(),
+                    )
+                    .split(f.size());
+
+                let info_layout = Layout::default()
+                    .direction(Direction::Horizontal)
+                    .constraints([Constraint::Percentage(100), Constraint::Min(30)].as_ref())
+                    .split(main_layout[0]);
+
+                f.render_widget(
+                    Paragraph::new(get_note_ascii(note_index).join("\n")),
+                    info_layout[1],
+                );
+
+                draw_bar_chart(f, main_layout[1], avg_db, cents);
+
+                draw_graph(f, main_layout[2], &state, &fft_buf, &cmndf, avg_f0);
+
+                f.render_widget(
+                    Paragraph::new(format!(
+                        "Input device: {}\n\
+                        Sample rate: {:.0}\n\
+                        Target frequency: {}\n\
+                        Detected frequency: {:.1}hz\n\
+                        ",
+                        device.name().unwrap(),
+                        state.sample_rate,
+                        match state.target_frequency {
+                            Some(f) => format!("{:.1}hz", f),
+                            None => "Not selected".into(),
+                        },
+                        avg_f0,
+                    )),
+                    info_layout[0],
+                );
+
+                let is_inside = state.cursor_pos.0 >= main_layout[2].top()
+                    && state.cursor_pos.0 < main_layout[2].bottom()
+                    && state.cursor_pos.1 >= main_layout[2].left()
+                    && state.cursor_pos.1 < main_layout[2].right();
+
+                if is_inside && state.mouse_pressed {
+                    let alpha = (state.cursor_pos.1 - main_layout[2].left()) as f32
+                        / (main_layout[2].width - 1) as f32;
+                    let (low, high) = match state.graph {
+                        Graph::AmplitudeSpectrum => (
+                            f32::ln(state.sample_rate / FFT_SIZE as f32),
+                            f32::ln(state.sample_rate / 2.0),
+                        ),
+                        Graph::CMNDF => (
+                            f32::ln(state.sample_rate / (cmndf.len() - 1) as f32),
+                            f32::ln(state.sample_rate / 2.0),
+                        ),
+                    };
```

Comment:
```suggestion
                    let low = match state.graph {
                        Graph::AmplitudeSpectrum => 
                            f32::ln(state.sample_rate / FFT_SIZE as f32),
                        Graph::CMNDF => 
                            f32::ln(state.sample_rate / (cmndf.len() - 1) as f32),
                    };
                    let high = f32::ln(state.sample_rate / 2.0);
```

It seems value `high` doesn't depend on `Graph`, so I guess it can be initialized outside of match